### ISSUE TITLE
Cannot use keyword await outside an async function `<script setup>`

### ISSUE THREAD

**Issue by ol1s** [vuejs/eslint-plugin-vue#1815]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.11
- **eslint-plugin-vue version:** 8.5
- **Node version:** 16
- **Operating System:** Linux

**Please show your full configuration:**
```
env:
  browser: true
  jest: true
  node: true
extends:
  - eslint:recommended
  - plugin:vue/vue3-recommended
plugins:
  - import
  - vue
```

**What did you do?**	

```vue
<script setup>
  const reports = ref([])
  reports.value = await getReports()
</script>
```

**What did you expect to happen?**

Expect no eslint errors for this snippet

**What actually happened?**
```
Parsing error: Cannot use keyword 'await' outside an async function
```

**Comment by ota-meshi** [comment]
Please see https://eslint.vuejs.org/user-guide/#using-eslint-v8-x
